addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -563,3 +563,63 @@ def test_gift_card_program_without_product(self):
         self._claim_reward(order, giftcard_program)
 
         self.assertEqual(giftcard_program.coupon_count, 0)
+
+    def test_apply_coupon_code_twice(self):
+        """This test ensures that applying a coupon with code twice will:
+            1. Raise an error
+            2. Not delete the coupon
+        """
+
+        # Create a coupon
+        coupon_program = self.env['loyalty.program'].create([
+            {
```

Comment:
```suggestion
        coupon_program = self.env['loyalty.program'].create({
```
single record creation, you can give a dict to the create call, instead of a list with one dict